src/components/new-safe/CreateSafe/useCreateSafe.ts

```diff
@@ -1,12 +1,30 @@
 import useIsWrongChain from '@/hooks/useIsWrongChain'
 import useWallet from '@/hooks/wallets/useWallet'
+import { useEffect } from 'react'
+import useLocalStorage from '@/services/local-storage/useLocalStorage'
+import type { StepRenderProps } from '@/components/new-safe/CardStepper/useCardStepper'
+import type { PendingSafeData } from '@/components/new-safe/steps/Step4'
+import type { NewSafeFormData } from '@/components/new-safe/CreateSafe/index'
+import { SAFE_PENDING_CREATION_STORAGE_KEY } from '@/components/new-safe/steps/Step4'
 
-const useCreateSafe = () => {
+const useCreateSafe = (setStep: StepRenderProps<NewSafeFormData>['setStep']) => {
+  const [pendingSafe] = useLocalStorage<PendingSafeData | undefined>(SAFE_PENDING_CREATION_STORAGE_KEY)
   const wallet = useWallet()
   const isWrongChain = useIsWrongChain()
 
   const isConnected = wallet && !isWrongChain
 
+  useEffect(() => {
```

Comment:
What do you think of separating this useEffect and the `isConnected` logic in two hooks? I suggest `useIsConnected` and `useSetCreationStep`

Reply:
Good idea, we could even reuse `useIsConnected` in other places. I also like @iamacook suggestion to have a context as to not having to include this hook in every component.

Reply:
I opted for using 2 hooks.